ui/src/app/views/workspace/component.ts

```diff
@@ -283,6 +292,38 @@ export class WorkspaceComponent implements OnInit, OnDestroy {
     this.router.navigate(['share'], {relativeTo : this.route});
   }
 
+  sendANotebook(notebook): void {
+    let enableBtn = false;
+    let unSelectCheckBox = false;
+    this.checkColumnNotebook = true;
+    this.enablePushNotebookBtn = false;
+
+    for (const file of this.notebookList) {
+      if (file.push === false) {
```

Comment:
This took me a while to understand, I might restructure this slightly:

```
let unselectCheckBox = false;
let enableBtn = false;

for(...) {
  if (file.push) {
    unselectCheckBox = true;
  } else {
    enableBtn = true;
  }
  if (unselectCheckBox && enableBtn) {
    break;
  }
}
this.checkColumnNotebook = !unselectCheckBox;
this.enablePushNotebookBtn = enableBtn;
```

Reply:
This logic was added to avoid having two for loops. Basically it checks if there is atleast one false checkbox then the group header checkbox is unselected. And if there exist atleast one true checkbox then Push button should be enabled I can rename the variables to make it more readable

Reply:
As mentioned over Slack / email: I don't think this UX is going to live for long. We should sync all notebooks to the cluster automatically, without asking users to check boxes and manually push individual notebooks.

If that's correct, this code may be going away soon, and I wouldn't worry too much about the details.

Reply:
Agreed on the UX. Even so, some fraction of this code will likely live on or will be copy-pasted going forward. It seems reasonable to review the code during code review, otherwise in my experience these kind of style issues are unlikely to ever be fixed (+ it serves as style learning for future PRs).

This was purely a coding comment. The current version of this is really verbose. I would just make two loops and favor readability, especially if this is a temporary thing.

Shorter version, replace this entire function with:
```
this.checkColumnNotebook = this.notebookList.some(f => !f.push);
this.enablePushNotebookBtn = this.notebookList.some(f => f.push);
```